### PR TITLE
Introdução ao map, filter, reduce

### PR DESCRIPTION
**Pull request by bluuesz**
Uma breve introdução as funções map, filter e reduce e alguns exemplos de como podem ser uteis e deixando o seu código mais ''enxuto'' e funcional evitando mutações e contendo funções puras.

### CODE REVIEW

**Comment by LeonardoCesca** [comment]
Hello @bluuesz,

Please, merge your branch with master branch.

Thank you for contribute to this project :)

**Comment by bluuesz** [comment]
Done, should I open a new pr?

**Comment by LeonardoCesca** [comment]
@bluuesz , 

You need run first `git pull origin master` and after `git push origin how-to-use-map-reduce-filter` to resolve conflicts

**Comment by bluuesz** [comment]
yes I did it.
looks like it does not worked

**Comment by brunormferreira** [comment]
Hellou devs,

I will go to solve the conflict, and adjust it.

Wait a minutes, plz.

Thanks,

**Comment by bluuesz** [comment]
I think it worked, sorry.

I wasn't giving ```git pull``` in the correct repository 😅

File: src/components/layout/layout.js [hunk 1]
```diff
@@ -79,8 +80,10 @@ const Layout = ({ children }) => {
                   </SidebarItems>
                   <SidebarItems to="/if-else/">If Else</SidebarItems>
                   <SidebarItems to="/for/">Loop For</SidebarItems>
+                  <SidebarItems to="/functional">Map, reduce, filter</SidebarItems>
```
Comment (brunormferreira) [inline src/components/layout/layout.js]: Change route name to `array-helpers` and his description to `Array Helpers`.

File: src/pages/functional/filter-with-map/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function MapFilter({ data }) {
+  const { mapWithFilter } = data.markdownRemark.frontmatter
+  return (
+    <Content>
+      <Title>{mapWithFilter[0].title}</Title>
+      <Paragraph>{mapWithFilter[1].subtitle}</Paragraph>
+      <Iframes
+        style={{  height: "5px" }}
+        description="iframe using combined operators"
```
Comment (brunormferreira) [inline src/pages/functional/filter-with-map/index.js]: Change description name

File: src/pages/functional/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  return (
+    <Content>
+      <Title>{title}</Title>
+      <Paragraph>{subtitle}</Paragraph>
+      <Title>Map();</Title>
+      <Paragraph>
+        A função map() funciona parecida como um loop for-each. Ele irá criar um novo array baseado no array e antigo e fazer as respectivas mudanças nele. Note que ele não faz nenhuma alteração no array original, vai ser criado um novo e retornado
+      </Paragraph>
+      <Iframes
+        description="iframe using If operator 1"
```
Comment (brunormferreira) [inline src/pages/functional/index.js]: Change description name

File: src/pages/functional/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        url="https://jsfiddle.net/bluues/yLzs4m8q/12/embedded/js/"
+      />
+
+      <Title>Filter();</Title>
+      <Paragraph>
+        Agora imagina que você está recebendo um array e o retorno dele vem algo do tipo ['Joao', 'José', undefined] Já pensou como seria chato tratar? Fazendo um for juntamente com alguns if's. Mas com o filter isso não é nescessario. Como o nome ja diz, ele vai filtrar apenas aqueles dados que o convêm, e funciona da mesma forma; Gerando um novo array de dados sem modificar o original.
+      </Paragraph>
+      <Iframes
+        description="iframe using If operator 2"
```
Comment (brunormferreira) [inline src/pages/functional/index.js]: Change description name

File: src/pages/functional/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+import Iframes from "../../components/iframes"
+
+export default function Functional({ data }) {
+  const {
+    title,
+    subtitle,
+    infos,
+    mapWithFilter
+
```
Comment (brunormferreira) [inline src/pages/functional/index.js]: Remove the white spaces

File: src/pages/functional/index.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      <ContentInfos>
+        <Title>{infos}</Title>
+        <Links>
+          <StyledLink to="/functional/filter-with-map">
+            {mapWithFilter[0].title}
+          </StyledLink>
+        </Links>
+      </ContentInfos>
+
```
Comment (brunormferreira) [inline src/pages/functional/index.js]: Remove the white spaces

**Pull request merged**